stream/stream_libarchive.c

```diff
@@ -244,8 +244,11 @@ struct mp_archive *mp_archive_new(struct mp_log *log, struct stream *src,
     struct mp_archive *mpa = talloc_zero(NULL, struct mp_archive);
     mpa->log = log;
     mpa->locale = newlocale(LC_ALL_MASK, "C.UTF-8", (locale_t)0);
```

Comment:
`LC_ALL_MASK` should be changed to `LC_CTYPE_MASK` here as well

Reply:
Why? The goal was to leave the default the same but just add a fallback.

Reply:
So `LC_ALL_MASK` isn't reliable on all systems from my testing, using `LC_CTYPE_MASK` should be all that's required for libarchive to function properly and appears to work on all systems.

Reply:
Well perhaps but that's outside the scope of this PR imo (since we wanted to be 100% sure that something else wouldn't break) and should be discussed elsewhere. Gonna mark this as resolved.

Reply:
Ok, I'll open a new PR for this.